static/src/javascripts/projects/common/modules/commercial/adblock-rules.js

```diff
@@ -0,0 +1,30 @@
+define([
+    'common/utils/config',
+    'common/utils/detect',
+    'common/utils/storage',
+    'common/modules/commercial/user-features'
+], function (
+    config,
+    detect,
+    storage,
+    userFeatures
+) {
+    function sharedRules() {
+        var alreadyVisted = storage.local.get('gu.alreadyVisited') || 0;
+
+        return detect.getBreakpoint() !== 'mobile' && detect.adblockInUse() && config.switches.adblock && alreadyVisted > 1;
+    }
+
+    function isMemberWithAdblock() {
+        return sharedRules() && userFeatures.isPayingMember();
+    }
+
+    function isNonMemberWithAdblock() {
+        return sharedRules() && !userFeatures.isPayingMember();
+    }
+
+    return {
+        isMemberWithAdblock: isMemberWithAdblock,
+        isNonMemberWithAdblock: isNonMemberWithAdblock
```

Comment:
I don't really think membership is a responsibility of this file. Why not just check the membership status in donot-use-adblock.js?